src/_common/content/content-editor/controls/mention/autocomplete/controls.ts

```diff
@@ -0,0 +1,223 @@
+		container: HTMLElement;
+		list: HTMLDivElement;
+	};
+
+	get displayUsers() {
+		if (this.isInverted) {
+			return this.users.slice().reverse();
+		}
+		return this.users;
+	}
+
+	get showControl() {
+		return this.visible && (this.isLoading || this.users.length > 0);
+	}
+
+	mounted() {
+		this.update();
+		document.addEventListener('keydown', this.onKeyDown);
+		this.isListening = true;
+	}
+
+	destroyed() {
+		if (this.isListening) {
+			document.removeEventListener('keydown', this.onKeyDown);
+			this.isListening = false;
+		}
+
+		if (this.remoteSuggestionTimeout) {
+			clearTimeout(this.remoteSuggestionTimeout);
+			this.remoteSuggestionTimeout = undefined;
+		}
+	}
+
+	@Watch('canShow')
+	@Watch('stateCounter')
+	private update() {
+		if (this.canShow && this.view instanceof EditorView) {
+			const state = this.view.state;
+			const node = ContentEditorService.getSelectedNode(this.view.state);
+
+			if (node !== null && node.isText) {
+				// Get all text from the beginning of the doc to the end of the selection.
+				// Then from the end of that slice, get the mention text.
+				const slice = state.doc.slice(0, state.selection.from);
+				const text = ContentEditorService.getFragmentText(slice.content);
+				const matches = /@([\w_-]+)$/i.exec(text);
+
+				if (matches && matches.length >= 2) {
+					const start = this.view.coordsAtPos(state.selection.from);
+
+					const box = this.$refs.container.offsetParent.getBoundingClientRect();
+					// If the text control is more than 50% down the page, open the control above ("inverted")
+					const relativeYPos = box.top / Screen.height;
+					if (relativeYPos >= 0.5) {
+						this.isInverted = true;
+						this.top = 'auto';
+						this.bottom = start.top - box.top + 30 + 'px';
+					} else {
+						this.isInverted = false;
+						this.top = start.top - box.top + 30 + 'px';
+						this.bottom = 'auto';
+					}
+
+					if (this.Screen.isXs) {
+						// On mobile, we want to position the element to the left border of the screen
+						this.left = '-' + box.left + 'px';
+					} else {
+						this.left = 'auto';
+					}
+
+					this.query = matches[1];
+					this.updateSuggestions(this.query);
+					this.visible = true;
+
+					return;
+				}
+			}
+		}
+
+		this.visible = false;
+	}
+
+	private async updateSuggestions(query: string) {
+		// Stop any existing queued search timeout.
+		if (this.remoteSuggestionTimeout) {
+			clearTimeout(this.remoteSuggestionTimeout);
+			this.remoteSuggestionTimeout = undefined;
+			this.isLoading = false;
+		}
+
+		this.selectedIndex = 0;
+		// If we already have the search results cached, use those and don't query backend.
+		if (ContentEditorMentionCache.hasResults(query)) {
+			this.users = ContentEditorMentionCache.getResults(query);
+			await this.handleInverted();
+		} else {
+			this.users = [];
+
+			// We set a timeout for 200ms here to not send right away when the user is typing fast.
+			this.remoteSuggestionTimeout = setTimeout(async () => {
```

Comment:
First line of the timeout function save the current instance of this.remoteSuggestionTImeout. We need to check if it changed after awaiting the suggestion request to see if we need to abort.

Reply:
This is related to your above comment, I'll resolve this.